source/slang/hlsl.meta.slang

```diff
@@ -2296,11 +2296,17 @@ ${{{{
         {
             auto componentName = componentNames[componentId];
             auto glslComponent = (isCmp ? "" :glslComponentNames[componentId == 0 ? 0 : componentId - 1]);
-    }}}}
+    
+            for (bool isStatus : { false, true })
+            {
+                const char* statusDecl = isStatus ? ", out uint status" : "";
+                const char* statusInit = isStatus ? " status = 0;\n" : "";
+}}}}
     [ForceInline]
     [require(glsl_hlsl_metal_spirv, texture_gather)]
-    vector<T,4> Gather$(cmp)$(componentName)($(samplerStateType)$(samplerStateParam) vector<float, Shape.dimensions+isArray> location $(cmpParam))
+    vector<T,4> Gather$(cmp)$(componentName)($(samplerStateType)$(samplerStateParam) vector<float, Shape.dimensions+isArray> location $(cmpParam) $(statusDecl))
     {
+        $(statusInit)
```

Comment:
I was looking into what `status` means, and it seems that this is to signify if "sparse memory accessed is resident"

The current implementations [needs to be changed](https://[registry.khronos.org/OpenGL/extensions/ARB/ARB_sparse_texture2.txt) to `sparseImage*` operations

Reply:
note: Even if we don't implement all sparse texture functions this PR, we should still remove `$(statusInit)` and use the correct SPIRV/GLSL function for `Gather` (with `status`) operations.

Reply:
Assigning `status` to zero is how the other functions are handling it.
https://github.com/shader-slang/slang/blob/323fdff4544ac3b640bdc617d7f7d64ab56fbf5a/source/slang/hlsl.meta.slang#L705

Removing `$(statusInit)` is easy when it is the actual implemented and tested properly.